djangocms_snippet/migrations/0010_cms4_grouper_version_data_migration.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import os
+
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+try:
+    from djangocms_versioning.constants import DRAFT
+    djangocsm_versioning_enabled = True
+except:
+    djangocsm_versioning_enabled = False
+
+
+def cms4_grouper_version_migration(apps, schema_editor):
+    grouper_count = 0
+    version_count = 0
+
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    Snippet = apps.get_model("djangocms_snippet", "Snippet")
+    SnippetGrouper = apps.get_model("djangocms_snippet", "SnippetGrouper")
+    User = apps.get_model('auth', 'User')
+
+    snippet_contenttype = ContentType.objects.get(app_label='djangocms_snippet', model='snippet')
+    snippet_queryset = Snippet.objects.all()
+
+    for snippet in snippet_queryset:
+        grouper = SnippetGrouper.objects.create()
+        snippet.new_snippet = grouper
+        snippet.save()
+        logger.info(f"Created Snippet Grouper ID: {snippet.snippet_grouper}")
+        grouper_count += 1
+
+        # Get a migration user.
+        migration_user = os.environ.get("DJANGOCMS_SNIPPET_VERSIONING_MIGRATION_USER_ID", None)
```

Comment:
This new setting will need to be a setting, we can then override the setting if it's not set in the aldryn_config.py file. 

The setting should be added in a new file: conf.py to keep the code consistent: https://github.com/django-cms/djangocms-moderation/blob/release/1.0.x/djangocms_moderation/conf.py